Data/Binary/Serialise/CBOR/ByteOrder.hs

```diff
@@ -210,6 +211,21 @@ wordToFloat64# w# =
           s'' ->
             case readDoubleArray# mba# 0# s'' of
               (# _, f# #) -> f#
+#else
+{-# INLINE wordToFloat64 #-}
+wordToFloat64 :: Word -> Double
+wordToFloat64 (W# w#) = D# (wordToFloat64# w#)
+
+{-# NOINLINE wordToFloat64# #-}
+wordToFloat64# :: Word# -> Double#
+wordToFloat64# w# =
+    case newByteArray# 8# realWorld# of
```

Comment:
@dcoutts Ah, suspect: We are allocating 8 bytes here, but we need four. Do you think the above is simply a waste of space or something semantically wrong?